executor/predictor/metadata.go

```diff
@@ -0,0 +1,122 @@
+package predictor
+
+import (
+	"fmt"
+	"github.com/seldonio/seldon-core/operator/apis/machinelearning.seldon.io/v1"
+)
+
+type MetadataTensor struct {
+	DataType string `json:"datatype,omitempty"`
+	Name     string `json:"name,omitempty"`
+	Shape    []int  `json:"shape,omitempty"`
+}
+
+type ModelMetadata struct {
+	Name     string           `json:"name,omitempty"`
+	Platform string           `json:"platform,omitempty"`
+	Versions []string         `json:"versions,omitempty"`
+	Inputs   []MetadataTensor `json:"inputs,omitempty"`
+	Outputs  []MetadataTensor `json:"outputs,omitempty"`
+}
+
+type GraphMetadata struct {
+	Name         string                   `json:"name,omitempty"`
+	Models       map[string]ModelMetadata `json:"models,omitempty"`
+	GraphInputs  []MetadataTensor         `json:"graphinputs,omitempty"`
+	GraphOutputs []MetadataTensor         `json:"graphoutputs,omitempty"`
+}
+
+func NewGraphMetadata(p *PredictorProcess, spec *v1.PredictorSpec) (output *GraphMetadata, err error) {
```

Comment:
Removal of named return parameters might make this function a little clearer to read